tests/miral/external_client.cpp

```diff
@@ -43,7 +45,7 @@ struct ExternalClient : miral::TestServer
     miral::X11Support x11_disabled_by_default{};
     miral::X11Support x11_enabled_by_default{miral::X11Support{}.default_to_enabled()};
 
-    std::string const output = tmpnam(nullptr);
+    std::string const output = mir::test::create_temp_file();
```

Comment:
This could be a `mir::Fd` from `mkstmp`; the `external_client.launch` lines would then redirect to the fd using `>&$FD`.